pygfx/renderers/wgpu/meshshader.py

```diff
@@ -193,27 +193,31 @@ def code_vertex(self):
         @stage(vertex)
         fn vs_main(in: VertexInput) -> Varyings {
 
+            // Get world transform
+            $$ if instanced
+                let instance_info = s_instance_infos[in.instance_index];
+                let world_transform = u_wobject.world_transform * instance_info.transform;
+            $$ else
+                let world_transform = u_wobject.world_transform;
+            $$ endif
+
             // Select what face we're at
             let index = i32(in.vertex_index);
             let face_index = index / 3;
             var sub_index = index % 3;
 
-            // If the camera flips a dimension, it flips the face winding.
-            // We can correct for this by adjusting the order (sub_index) here.
-            sub_index = select(sub_index, -1 * (sub_index - 1) + 1, u_stdinfo.flipped_winding > 0);
+            // If a transform has an uneven number of negative scales, the 3 vertices
+            // that make up the face are such that the GPU will mix up front and back
+            // faces, producing an incorrect is_front. We can detect this from the
+            // sign of the determinant, and reorder the faces to fix it.
+            let winding_world = sign(determinant(world_transform));
+            let winding_cam = sign(determinant(u_stdinfo.cam_transform));
```

Comment:
Does someone with better Math skills happen to know a more efficient way to calculate the sign of the determinant of a 4x4? Maybe @FirefoxMetzger ?

Reply:
That depends on what info we have access to without doing any extra work. 

- If you have an arbitrary 4x4 matrix (`A`), `determinant(A)` is about as good as it gets. That is assuming that it is implemented using something like an LU decomposition internally or comes from a trusted source.
- If you have a transformation matrix in standard form (bottom row is `(0, 0, 0, 1)`) you can focus on `A[:3, :3]` and compute the determinant of that 3x3 matrix instead. This makes it slightly faster. There might even be a specialized algorithm available, but I would have to go and search for it.
- If you have the parts of the transform (translation, rotation, scale), then you should be able to simply do `winding_cam = prod(sign(scale))` and ignore translation and rotation (both always have `det(A)=1`). I.e., you can count the negative scale factors and if there is an uneven number you have `det(A)<0`.

Reply:
This is what Three does: "compute the determinant of that 3x3 matrix"

Reply:
Thanks for the suggestions. I did the 3x3 trick.